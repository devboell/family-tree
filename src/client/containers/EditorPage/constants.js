export const editModes = {
  DISABLED: 'disabled',
  CREATE: 'create',
  UPDATE: 'update',
}


export const SET_EDIT_MODE = 'SET_EDIT_MODE'
export const SET_SELECTED_PERSON = 'SET_SELECTED_PERSON'
