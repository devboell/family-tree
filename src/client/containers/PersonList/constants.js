export default {
  SELECTED: 'SELECTED',
  SELECT_FIRST: 'SELECT_FIRST',
  UNSELECTED: 'UNSELECTED',
}
