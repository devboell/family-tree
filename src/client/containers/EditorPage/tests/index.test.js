import React from 'react'
import { EditorPage } from '../index'
import { editModes } from '../constants'

const relationsQuery = {
  loading: false,
  relations: [],
}

const personsQuery = {
  loading: false,
  persons: [],
}

const props = {
  relationsQuery,
  personsQuery,
  createPersonMutation: jest.fn(),
  updatePersonMutation: jest.fn(),
  removePersonMutation: jest.fn(),
  editMode: editModes.DISABLED,
  onSelectPerson: jest.fn(),
  onCreatePerson: jest.fn(),
  onRemovePerson: jest.fn(),
  selectedPerson: {},
}

it('EditorPage, renders the component', () => {
  const wrapper = shallow(<EditorPage {...props} />)
  expect(wrapper).toMatchSnapshot()
})
