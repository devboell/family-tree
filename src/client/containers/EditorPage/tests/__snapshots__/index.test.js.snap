// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditorPage, renders the component 1`] = `
<styled.div>
  <PersonList
    onSelectPerson={[Function]}
    persons={Array []}
    selectedPerson={Object {}}
  />
  <styled.div>
    <EditorControls
      editMode="disabled"
      onCreatePerson={[Function]}
      onRemovePerson={[Function]}
      removePersonMutation={[Function]}
      selectedPerson={Object {}}
    />
    <ReduxForm
      disabled={true}
      initialValues={Object {}}
      onSubmit={[Function]}
      persons={Array []}
      relations={Array []}
    />
  </styled.div>
</styled.div>
`;
